librz/search/collection.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_search.h>
+#include "search_internal.h"
+
+/**
+ * \brief      Initialize a new RzSearchCollection
+ *
+ * \param[in]  find      The find callback to set
+ * \param[in]  is_empty  The callback to use to check if collection is empty
+ * \param[in]  free      The callback to use to free the context
+ * \param      user      The additional context needed.
+ *
+ * \return     On success returns a valid pointer, otherwise NULL.
+ */
+RZ_IPI RZ_OWN RzSearchCollection *rz_search_collection_new(RZ_NONNULL RzSearchFindCallback find, RZ_NONNULL RzSearchIsEmptyCallback is_empty, RZ_NULLABLE RzSearchFreeCallback free, RZ_NULLABLE void *user) {
```

Comment:
Interesting concept. Probably could be generalized and put into RzUtil. But that could be done after this PR

Reply:
yes, it could be.